plugins/woocommerce/src/Internal/Admin/Orders/ListTable.php

```diff
@@ -649,4 +654,141 @@ private function print_hidden_form_fields(): void {
+			exit;
+		}
+
+		$report_action = '';
+		$changed       = 0;
+
+		if ( 'remove_personal_data' === $action ) {
+			$report_action = 'removed_personal_data';
+			$changed       = $this->do_bulk_action_remove_personal_data( $ids );
+		} elseif ( false !== strpos( $action, 'mark_' ) ) {
+			$order_statuses = wc_get_order_statuses();
+			$new_status     = substr( $action, 5 );
+			$report_action  = 'marked_' . $new_status;
+
+			if ( isset( $order_statuses[ 'wc-' . $new_status ] ) ) {
+				$changed = $this->do_bulk_action_mark_orders( $ids, $new_status );
+			}
+		}
+
+		if ( $changed ) {
+			$redirect_to = add_query_arg(
+				array(
+					'bulk_action' => $report_action,
+					'changed'     => $changed,
+					'ids'         => implode( ',', $ids ),
+				),
+				$redirect_to
+			);
+		}
+
+		wp_safe_redirect( $redirect_to );
+		exit;
+	}
+
+	/**
+	 * Implements the "remove personal data" bulk action.
+	 *
+	 * @param array $order_ids The Order IDs.
+	 * @return int Number of orders modified.
+	 */
+	private function do_bulk_action_remove_personal_data( $order_ids ): int {
+		$changed = 0;
+
+		foreach ( $order_ids as $id ) {
+			$order = wc_get_order( $id );
+
+			if ( $order ) {
+				do_action( 'woocommerce_remove_order_personal_data', $order ); // phpcs:ignore WooCommerce.Commenting.CommentHooks.MissingHookComment
+				$changed++;
+			}
+		}
+
+		return $changed;
+	}
+
+	/**
+	 * Implements the "mark <status>" bulk action.
+	 *
+	 * @param array  $order_ids  The order IDs to change.
+	 * @param string $new_status The new order status.
+	 * @return int Number of orders modified.
+	 */
+	private function do_bulk_action_mark_orders( $order_ids, $new_status ): int {
+		$changed = 0;
+
+		// Initialize payment gateways in case order has hooked status transition actions.
+		WC()->payment_gateways();
+
+		foreach ( $order_ids as $id ) {
+			$order = wc_get_order( $id );
+			$order->update_status( $new_status, __( 'Order status changed by bulk edit:', 'woocommerce' ), true );
+			do_action( 'woocommerce_order_edit_status', $id, $new_status ); // phpcs:ignore WooCommerce.Commenting.CommentHooks.MissingHookComment
+			$changed++;
+		}
+
+		return $changed;
+	}
+
+	/**
+	 * Show confirmation message that order status changed for number of orders.
+	 */
+	public function bulk_action_notices() {
+		if ( empty( $_REQUEST['bulk_action'] ) ) {
+			return;
+		}
+
+		$order_statuses = wc_get_order_statuses();
+		$number         = isset( $_REQUEST['changed'] ) ? absint( $_REQUEST['changed'] ) : 0;
+		$bulk_action    = wc_clean( wp_unslash( $_REQUEST['bulk_action'] ) );
+
+		// Check if any status changes happened.
+		foreach ( $order_statuses as $slug => $name ) {
+			if ( 'marked_' . str_replace( 'wc-', '', $slug ) === $bulk_action ) { // WPCS: input var ok, CSRF ok.
+				/* translators: %d: orders count */
+				$message = sprintf( _n( '%d order status changed.', '%d order statuses changed.', $number, 'woocommerce' ), number_format_i18n( $number ) );
```

Comment:
```suggestion
				/* translators: %s: orders count */
				$message = sprintf( _n( '%s order status changed.', '%s order statuses changed.', $number, 'woocommerce' ), number_format_i18n( $number ) );
```

I appreciate it's unlikely we'll be dealing with thousands of orders in a single bulk operation ... *but* ... `%s` is probably preferable since `number_format_i18n()` returns a formatted string complete with thousands separators (and it's filterable) which when cast to integer may not reflect the actual value.

For instance, currently, if `$number` is `1000` the message will say `'1 order statuses changed.'`.